docs/docs/external-volumes.md

```diff
@@ -172,15 +172,20 @@ The default implicit volume size is 16 GB. If you are using the Mesos containeri
 
 ### Potential Pitfalls
 
+*   If one or more external volumes are declared for a Marathon app, and the Docker image specification includes one or more `VOLUME` entries, Docker may create anonymous external volumes. This is default Docker behavior with respect to volume management when the `--volume-driver` flag is passed to `docker run`. Follow these steps to prevent Docker from creating anonymous volumes:
```

Comment:
So we're talking about external volumes in this section and the default
implementation, rexray. The reference implementation that we offer (via AWS
templates) integrates with EBS. Also realize that neither dc/os or marathon
offer volume lifecycle management features for external volumes. Meaning
that Dan (or Alice) needs keep track of all volumes used by apps and delete
old/expired volumes when they're no longer needed.

To answer your question: when Docker creates anonymous external volumes for
an app (and doesn't subsequently delete them (expected behavior)) a couple
of things happen:
* The anonymous volumes linger on in the storage system (on AWS this means
lots of EBS volumes accumulating over time)
* Dan (or Alice) need to identify these volumes and delete them; this is
relatively straight forward given the obvious UUID nature of the names but
the volumes created this way aren't labeled by a mechanism that makes them
easy to select (and delete) in bulk.
* Since it's not immediately clear that Docker is doing this, Dan (or
Alice) may be (a) overwhelmed by suddenly needing to manage a large number
of volumes, and; (b) surprised by a very large AWS bill at the end of the
month because they weren't tracking all the anonymous volumes that Docker
created and suddenly are using 30TB of space (in the form of detached,
anonymous volumes) on EBS.


On Mon, Jul 25, 2016 at 1:56 PM, Suzanne Scala <notifications@github.com>
wrote:

> In docs/docs/external-volumes.md
> <https://github.com/mesosphere/marathon/pull/4131#discussion_r72111953>:
>
> > @@ -172,15 +172,20 @@ The default implicit volume size is 16 GB. If you are using the Mesos containeri
> >
> >  ### Potential Pitfalls
> >
> > +*   If one or more external volumes are declared for a Marathon app, and the Docker image specification includes one or more `VOLUME` entries, Docker may create anonymous external volumes. This is default Docker behavior with respect to volume management when the `--volume-driver` flag is passed to `docker run`. Follow these steps to prevent Docker from creating anonymous volumes:
>
> Why wouldn't I want Docker to create anonymous volumes?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4131/files/fe790975b35e9e54a4d52b6d45b89d79a87f63ca#r72111953>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLG_0YuxS3BvBiCtj8r220Xwd-lnTks5qZPi0gaJpZM4JTHxm>
> .
>